src/main/java/seedu/address/logic/commands/schedule/ImportCommand.java

```diff
@@ -0,0 +1,132 @@
+package seedu.address.logic.commands.schedule;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_FILEPATH;
+import static seedu.address.model.Model.PREDICATE_SHOW_ALL_PERSONS;
+
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.exceptions.DataConversionException;
+import seedu.address.commons.exceptions.IllegalValueException;
+import seedu.address.commons.util.JsonUtil;
+import seedu.address.logic.EditUtil.EditPersonDescriptor;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.EditTypeCommand;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+import seedu.address.model.schedule.Schedule;
+import seedu.address.storage.JsonAdaptedSchedule;
+
+
+/**
+ * Imports a {@code Schedule} from the specified json file and replaces the indexed {@code Person}'s schedule with
+ * the imported {@code Schedule}
+ */
+public class ImportCommand extends EditTypeCommand {
+
+    public static final String COMMAND_WORD = "import";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Imports a schedule from a json file and "
+            + "replaces the indexed person's schedule with the imported schedule.\n"
+            + "Parameters: PERSON_INDEX (must be a positive integer) "
+            + PREFIX_FILEPATH + "FILEPATH\n"
+            + "Example: " + COMMAND_WORD + " 1 " + PREFIX_FILEPATH + "typicalSchedule.json";
+
+    public static final String MESSAGE_SUCCESS = "%1$s's schedule has been replaced with the imported schedule!";
+    public static final String MESSAGE_FAILURE =
+            "Data in %s is in invalid format, %s's schedule is unchanged.";
```

Comment:
```suggestion
    public static final String MESSAGE_FAILURE = "Data in %s is in invalid format, %s's schedule is unchanged.";
```
Did not exceed character limit